src/checkers/inference/util/CopyUtil.java

```diff
@@ -166,7 +166,7 @@ private static void copyAnnotationsImpl(final AnnotatedTypeMirror from, final An
              // Primitives only take one annotation, which was already copied
 
         } else if (fromKind == NONE || fromKind == NULL || fromKind == VOID ||
-                toKind == NONE || toKind == NULL || fromKind == NULL) {
```

Comment:
For symmetry, should this be `toKind == VOID`? How can from and to kinds not match?